api/src/test/java/org/openmrs/api/impl/ConditionServiceImplTest.java

```diff
@@ -0,0 +1,183 @@
+		Assert.assertEquals(ConditionVerificationStatus.CONFIRMED, savedCondition.getVerificationStatus());
+		Assert.assertEquals(ConditionClinicalStatus.ACTIVE, savedCondition.getClinicalStatus());
+		Assert.assertEquals(uuid, savedCondition.getUuid());
+		Assert.assertEquals(patientId, savedCondition.getPatient().getPatientId());
+		Assert.assertEquals(codedOrFreeText, savedCondition.getCondition());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldFindConditionGivenValidUuid() {
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		String clinical_status = "INACTIVE";
+		String verification_status = "PROVISIONAL";
+		Condition condition = conditionService.getConditionByUuid(uuid);
+		Assert.assertEquals(clinical_status, condition.getClinicalStatus().toString());
+		Assert.assertEquals(verification_status, condition.getVerificationStatus().toString());
+	}
+
+	/**
+	 * @see ConditionService#getConditionByUuid(String)
+	 */
+	@Test
+	public void getConditionByUuid_shouldReturnNullIfNoObjectFoundWithGivenUuid() {
+		Assert.assertNull(conditionService.getConditionByUuid("invalid uuid"));
+	}
+
+	/**
+	 * @see ConditionService#getCondition(Integer) 
+	 */
+	@Test
+	public void getConditionById_shouldFindConditionGivenValidId() {
+		int conditionId = 1;
+		String uuid="2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		String clinical_status = "INACTIVE";
+		String verification_status = "PROVISIONAL";
+		Condition condition = conditionService.getCondition(conditionId);
+		Assert.assertEquals(clinical_status, condition.getClinicalStatus().toString());
+		Assert.assertEquals(verification_status, condition.getVerificationStatus().toString());
+		Assert.assertEquals(uuid, condition.getUuid());
+	}
+
+	/**
+	 * @see ConditionService#getActiveConditions(Patient)
+	 */
+	@Test
+	public void getActiveConditions_shouldGetActiveConditions() {
+		int patientId = 2;
+		List<Condition> activeConditions = conditionService.getActiveConditions(patientService.getPatient(patientId));
+		Condition firstActiveCondition = activeConditions.get(0);
+		Condition secondActiveCondition = activeConditions.get(1);
+		String firstUUID = "2cc6880e-2c46-11e4-9138-a6c5e4d20fb7";
+		String secondUUID = "2cc6880e-2c46-15e4-9038-a6c5e4d22fb7";
+		Assert.assertFalse(activeConditions.isEmpty());
+		Assert.assertTrue(activeConditions.size() == 2);
+		Assert.assertEquals(firstUUID,firstActiveCondition.getUuid());
+		Assert.assertEquals(secondUUID, secondActiveCondition.getUuid());
+	}
+	
+	
+
+	/**
+	 * @see ConditionService#voidCondition(Condition, String) 
+	 */
+	@Test
+	public void voidCondition_shouldVoidConditionSuccessfully(){
+		String voidReason = "Test Reason";
+		String uuid = "2cc6880e-2c46-11e4-9138-a6c5e4d20fb7";
+		Condition nonVoidedCondition = conditionService.getCondition(2);
+		Condition voidedCondition = conditionService.voidCondition(nonVoidedCondition, voidReason);
```

Comment:
You need sanity checks before this call to ensure that voided is set to false and the other void related fields are null

Reply:
Okay. Done